build-scripts/gulp/hassio.js

```diff
@@ -1,13 +1,33 @@
 const gulp = require("gulp");
+const fs = require("fs");
+const path = require("path");
 
 const env = require("../env");
+const paths = require("../paths");
 
 require("./clean.js");
 require("./gen-icons-json.js");
 require("./webpack.js");
 require("./compress.js");
 require("./rollup.js");
 
+async function backwardsCompatibility() {
+  // Until Supervisor 228, Home Assistant expected the entrypoint
+  // to be available at /api/hassio/app/entrypoint.js
+  // Then we changed to two builds, so now we have 2 folders.
+  //
+  // We redirect to the ES5 build here.
+  fs.writeFileSync(
+    path.resolve(paths.hassio_output_root, "entrypoint.js"),
+    `
+var el = document.createElement('script');
```

Comment:
Wouldn't a 301 be better?

Reply:
We can't. This is also running in old versions of HA and so we can't change the hassio integration to register a redirect.

Reply:
Oh right.